nupic/research/frameworks/dendrites/dendritic_layers.py

```diff
@@ -107,3 +108,72 @@ def apply_dendrites(self, y, dendrite_activations):
         return y * torch.sigmoid(
             torch.abs(dendrite_activations).max(dim=2).values * sign_mask
         )
+
+
+class AbsoluteMaxGatingDendriticLayer2d(SparseWeights2d):
+    """
+    A dendrite layer using the absolute max gating formulation, for convolutional
+    layers. The multiplicative dendrite outputs are applied element-wise to each output
+    channel. That is, for a given output channel, all activation values (determined by
+    the convolution operation) are multiplied by a single value computed via dendrites.
+    """
+
+    def __init__(
+        self, module, num_segments, dim_context,
+        module_sparsity, dendrite_sparsity, dendrite_bias=None
+    ):
+        """
+        :param module: conv2d module which performs the forward pass
+        :param num_segments: number of dendrite segments per out-unit
+        :param dim_context: length of the context vector;
+                            the same context will be applied to each segment
+        :param module_sparsity: sparsity applied over linear module;
+        :param dendrite_sparsity: sparsity applied transformation per unit per segment
+        :param dendrite_bias: whether or not dendrite activations have an additive bias
+        """
+        self.segments = None
+        super().__init__(module, sparsity=module_sparsity)
+
+        self.segments = DendriteSegments(
+            num_units=module.out_channels,
+            num_segments=num_segments,
+            dim_context=dim_context,
+            sparsity=dendrite_sparsity,
+            bias=dendrite_bias,
+        )
+
+        self.rezero_weights()
+
+    def rezero_weights(self):
+        super().rezero_weights()
+        if self.segments is not None:  # only none at beggining of init
+            self.segments.rezero_weights()
+
+    def apply_dendrites(self, y, dendrite_activations):
+        output_max = dendrite_activations.max(dim=2).values
```

Comment:
I wonder if this can be done in a simpler way. Possibly, you can call `idices = dendrite_activations.abs().max(dim=2).indices` and then unpack the indices as appropriate.

Reply:
Yes, just made the change!